src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewController.swift

```diff
@@ -31,43 +31,54 @@ class ExposureSubmissionWarnOthersViewController: DynamicTableViewController, EN
 		super.viewDidLoad()
 
 		setupView()
+		footerView?.isHidden = false
 	}
 
-	// MARK: - Protocol ENANavigationControllerWithFooterChild
+	override var navigationItem: UINavigationItem {
+		navigationFooterItem
+	}
 
-	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
-		onPrimaryButtonTap { [weak self] isLoading in
-			DispatchQueue.main.async {
-				self?.navigationFooterItem?.isPrimaryButtonLoading = isLoading
-				self?.navigationFooterItem?.isPrimaryButtonEnabled = !isLoading
-			}
-		}
```

Comment:
Why remove this here?

Reply:
As in `ExposureSubmissionQRInfoViewController`: code style. By moving everything protocol-related into a dedicated extension you can move out or refactor protocol implementations way faster and easier than looking up all functions in the current MARK-nonsense.

Reply:
But reading the code is more complex if you open up the file you can't immediately see which protocols are implemented. You have to search all over the project for possible extensions. I also don't see why refactoring should be faster that way? Copying from an extensions section is basically the same effort as copying from a Mark section. 
Also this file was change just before weeks to the MARK-nonsense (how you call it) and now it gets changed back again. As agreed in the dev camp we decided to use MARKs. If you like to work in a different style, make a request and we can vote on what the majority wants.

Reply:
> But reading the code is more complex if you open up the file you can't immediately see which protocols are implemented. You have to search all over the project for possible extensions. 

You don't read the code like a book, don't you? If you check for references, symbols or even text-matches you'll end at the correct place, even if further down in the code.

> I also don't see why refactoring should be faster that way? Copying from an extensions section is basically the same effort as copying from a Mark section.

If I need some private helper functions I have to clutter them into other sections not related to the protocol implementation.

The only downside might come from very large protocols but monstrosities like `ExposureSubmissionService` should be split up in future. We can structure code like in `AppConfigurationProviding` and `AppConfigurationFetching` and even group it like in `Store`. 

Request for discussion is open.